lib/galaxy/jobs/command_factory.py

```diff
@@ -19,6 +19,13 @@
 SETUP_GALAXY_FOR_METADATA = """
 [ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
 """
+PREPARE_DIRS = """mkdir -p working outputs
+if [ -f .job_started ]; then
+    rm -rf working/ outputs/; cp -R _working working; cp -R _outputs outputs
+else
+    cp -R working _working; cp -R outputs _outputs
```

Comment:
Add a random suffix to prevent collision?

Reply:
Something time-derived like we do for the tmp (and home?) dirs would be preferable than random.

Reply:
what would be the scenario for a collision ?

Reply:
Retried twice? I guess... those are cps and mvs though - maybe I didn't look at this close enough before responding...

Reply:
The idea is that we'd always remove `working/` and `outputs/` and then copy their "backups" when the job previously created `.job_started`, so I think it should be safe ?